scrapy/extensions/feedexport.py

```diff
@@ -524,17 +524,31 @@ def build_instance(builder, *preargs):
             raise TypeError(f"{feedcls.__qualname__}.{method_name} returned None")
         return instance
 
-    def _get_uri_params(self, spider, uri_params, slot=None):
+    def _get_uri_params(
+        self,
+        spider: Spider,
+        uri_params_function: Optional[Union[str, Callable[[dict, Spider], dict]]],
```

Comment:
`Union[str, Callable[[dict, Spider], dict], None]`?

Reply:
I would think `Optional[Union[…]]` is more Pythonic than `Union[…, None]`, even if longer. I don’t really know, though.